src/fu-util.c

```diff
@@ -727,6 +732,62 @@ fu_util_report_history_for_uri (FuUtilPrivate *priv,
 				  SOUP_MEMORY_COPY, data, strlen (data));
 	status_code = soup_session_send_message (priv->soup_session, msg);
 	g_debug ("server returned: %s", msg->response_body->data);
+
+	/* parse JSON reply */
+	json_parser = json_parser_new ();
+	if (!json_parser_load_from_data (json_parser,
+					 msg->response_body->data,
+					 msg->response_body->length,
+					 error)) {
+		g_prefix_error (error, "Failed to parse JSON response: ");
+		return FALSE;
+	}
+	json_root = json_parser_get_root (json_parser);
+	if (json_root == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_PERMISSION_DENIED,
+				     "JSON response was malformed");
+		return FALSE;
+	}
+	json_object = json_node_get_object (json_root);
+	if (json_object == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_PERMISSION_DENIED,
+				     "JSON response object was malformed");
+		return FALSE;
+	}
+
+	/* get any optional server message */
+	if (json_object_has_member (json_object, "msg"))
+		server_msg = json_object_get_string_member (json_object, "msg");
+
+	/* server reported failed */
+	if (!json_object_get_boolean_member (json_object, "success")) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_PERMISSION_DENIED,
+			     "Server rejected report: %s",
+			     server_msg != NULL ? server_msg : "unspecified");
+		return FALSE;
+	}
+
+	/* server wanted us to see the message */
+	if (server_msg != NULL) {
+		if (g_strcmp0 (server_msg, "known issue") == 0 &&
+		    json_object_has_member (json_object, "uri")) {
+			g_print ("%s %s\n",
+				 /* TRANSLATORS: the server sent the user a small message */
+				 _("Update failure is a known issue, visit this URL for more information:"),
+				 json_object_get_string_member (json_object, "uri"));
```

Comment:
Perhaps offer to xdg-open it at this point with a prompt?

Reply:
I did ponder that; I think the user wouldn't expect that from a console program. From gnome-software, sure, just not here.

Reply:
Gotcha, makes sense.